### DISCUSSION TITLE
How to run a flask app in the background , while my entry point file is another file?

### DISCUSSION THREAD

**Discussion by Agrover112** [EddieHubCommunity/support#395] [Q&A] [answered]
I'm using streamlit for making the front-end therefore, the entry point is a Streamlit file. 
However, I would like to to start a Flask API in the background so that one of the files associated with Streamlit can send some request and receive data.

**Comment by eddiejaoude** [comment]
I will share in the python channel in our Discord too 🤓

**Comment by RemcoHalman** [comment]
> I'm using streamlit for making the front-end therefore, the entry point is a Streamlit file.
> However, I would like to to start a Flask API in the background so that one of the files associated with Streamlit can send some request and receive data.

Where are you hosting the project? Is it on a server or somewhere else? Have done some Flask before so perhaps with a little more info i would know an answer.

**Comment by Agrover112** [comment, marked as answer]
I found my solution. Instead of using UWSGI default provided by Flask which is single threaded anyway.
I deployed it locally on a port using waitress , then one can use pythonw and set is as a background running startup process at the given port!.
Alternative is to use tmux or screen or python app.py & to run it in the background

**Reply by eddiejaoude** [reply]
Thanks for sharing your solution @Agrover112 👍
